apps/fortuna/src/history.rs

```diff
@@ -0,0 +1,437 @@
+use crate::api::ChainId;
+use chrono::{DateTime, NaiveDateTime};
+use ethers::core::utils::hex::ToHex;
+use ethers::prelude::TxHash;
+use ethers::types::Address;
+use serde::Serialize;
+use sqlx::{migrate, Pool, Sqlite, SqlitePool};
+use std::sync::Arc;
+use tokio::spawn;
+use tokio::sync::mpsc;
+use utoipa::ToSchema;
+
+#[derive(Clone, Debug, Serialize, ToSchema, PartialEq)]
+pub enum RequestEntryState {
+    Pending,
+    Completed {
+        reveal_block_number: u64,
+        reveal_tx_hash: TxHash,
+    },
+    Failed {
+        reason: String,
+    },
+}
+
+#[derive(Clone, Debug, Serialize, ToSchema, PartialEq)]
+pub struct RequestStatus {
+    pub chain_id: ChainId,
+    pub sequence: u64,
+    pub created_at: DateTime<chrono::Utc>,
+    pub last_updated_at: DateTime<chrono::Utc>,
+    pub request_block_number: u64,
+    pub request_tx_hash: TxHash,
+    pub sender: Address,
+    pub state: RequestEntryState,
+}
+
+#[derive(Clone, Debug, Serialize, ToSchema, PartialEq)]
+struct RequestRow {
+    chain_id: String,
+    sequence: i64,
+    created_at: NaiveDateTime,
+    last_updated_at: NaiveDateTime,
+    state: String,
+    request_block_number: i64,
+    request_tx_hash: String,
+    sender: String,
+    reveal_block_number: Option<i64>,
+    reveal_tx_hash: Option<String>,
+    info: Option<String>,
+}
+
```

Comment:
this code has a bunch of panics and unwraps. we shouldn't do that -- we don't want this code to crash the keeper service

Reply:
(literally just had a price feeds outage caused by a panic in an off-chain service)